Build instructions not setting up the necesary include and library paths for install

Hi all. Love the project but hit a stumble on installation. 
Im on linux. I cloned USD from github so the USD include dir and USD library dir live in $HOME/dev/USD/build and are not on any default search paths.
Following the build instructions I pass my USD_ROOT to cmake and it generates the Makefile fine. 
However, when I run 
`cmake --build  . -- VERBOSE=1 -j8 all test install`
or when i run
`make`
I get compiler errors complaining that it cant find the USD header files.
So I can fix that by setting the env variable CPLUS_INCLUDE_PATH to the usd include dir.
Then I get linking errors though. Which I can fix by setting LIBRARY_PATH to the usd library dir.
I'm not familiar with how cmake works at all really, but since I provide it with my the root of my USD build, and I can see in CMakeCache.txt that it does correctly have the USD_INCLUDE_DIR and USD_LIBRARY_DIR set, shouldn't it also be able to set these up to be used by the Makefile for building?

Just to be clear these are the commands I was running originally and the compilation error:
```
michaelb@boswell:~/dev/USDPluginExamples/build$ rm -r *
michaelb@boswell:~/dev/USDPluginExamples/build$ ls .
michaelb@boswell:~/dev/USDPluginExamples/build$ ls ..
build  cmake  CMakeLists.txt  images  LICENSE  README.md  src
michaelb@boswell:~/dev/USDPluginExamples/build$ echo $USD
/home/michaelb/dev/USD/build
michaelb@boswell:~/dev/USDPluginExamples/build$ cmake -DUSD_ROOT=$USD .. 
-- The CXX compiler identification is GNU 9.2.1
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython2.7.so (found suitable version "2.7.17", minimum required is "2.7") 
-- Found PythonInterp: /usr/bin/python2.7 (found suitable version "2.7.17", minimum required is "2.7") 
-- Found Boost: /usr/include (found version "1.67.0")  
-- Found Boost: /usr/include (found version "1.67.0") found components: python27 
-- /home/michaelb/dev/USD/build
-- Found USD: /home/michaelb/dev/USD/build/include (found version "0.20.11") 
-- Found TBB: /usr/include (found version "2019.0")  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/michaelb/dev/USDPluginExamples/build
michaelb@boswell:~/dev/USDPluginExamples/build$ make
Scanning dependencies of target usdTri
[  4%] Building CXX object src/usdTri/CMakeFiles/usdTri.dir/tokens.cpp.o
In file included from /home/michaelb/dev/USDPluginExamples/src/usdTri/tokens.cpp:6:
/home/michaelb/dev/USDPluginExamples/src/usdTri/./tokens.h:18:10: fatal error: pxr/pxr.h: No such file or directory
   18 | #include "pxr/pxr.h"
      |          ^~~~~~~~~~~
compilation terminated.
make[2]: *** [src/usdTri/CMakeFiles/usdTri.dir/build.make:83: src/usdTri/CMakeFiles/usdTri.dir/tokens.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:224: src/usdTri/CMakeFiles/usdTri.dir/all] Error 2
make: *** [Makefile:150: all] Error 2
```
Changing that cmake line to 
`cmake -DUSD_ROOT=$USD -DTBB_ROOT=$USD -DBOOST_ROOT=$USD .`
allows the make command to run successfully.

Sure thing! 
```
michaelb@boswell:~/dev/USDPluginExamples/build$ make VERBOSE=1
/usr/local/bin/cmake -S/home/michaelb/dev/USDPluginExamples -B/home/michaelb/dev/USDPluginExamples/build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/local/bin/cmake -E cmake_progress_start /home/michaelb/dev/USDPluginExamples/build/CMakeFiles /home/michaelb/dev/USDPluginExamples/build/CMakeFiles/progress.marks
make  -f CMakeFiles/Makefile2 all
make[1]: Entering directory '/home/michaelb/dev/USDPluginExamples/build'
make  -f src/usdTri/CMakeFiles/usdTri.dir/build.make src/usdTri/CMakeFiles/usdTri.dir/depend
make[2]: Entering directory '/home/michaelb/dev/USDPluginExamples/build'
cd /home/michaelb/dev/USDPluginExamples/build && /usr/local/bin/cmake -E cmake_depends "Unix Makefiles" /home/michaelb/dev/USDPluginExamples /home/michaelb/dev/USDPluginExamples/src/usdTri /home/michaelb/dev/USDPluginExamples/build /home/michaelb/dev/USDPluginExamples/build/src/usdTri /home/michaelb/dev/USDPluginExamples/build/src/usdTri/CMakeFiles/usdTri.dir/DependInfo.cmake --color=
Dependee "/home/michaelb/dev/USDPluginExamples/build/src/usdTri/CMakeFiles/usdTri.dir/DependInfo.cmake" is newer than depender "/home/michaelb/dev/USDPluginExamples/build/src/usdTri/CMakeFiles/usdTri.dir/depend.internal".
Dependee "/home/michaelb/dev/USDPluginExamples/build/src/usdTri/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/home/michaelb/dev/USDPluginExamples/build/src/usdTri/CMakeFiles/usdTri.dir/depend.internal".
Scanning dependencies of target usdTri
make[2]: Leaving directory '/home/michaelb/dev/USDPluginExamples/build'
make  -f src/usdTri/CMakeFiles/usdTri.dir/build.make src/usdTri/CMakeFiles/usdTri.dir/build
make[2]: Entering directory '/home/michaelb/dev/USDPluginExamples/build'
[  4%] Building CXX object src/usdTri/CMakeFiles/usdTri.dir/tokens.cpp.o
cd /home/michaelb/dev/USDPluginExamples/build/src/usdTri && /usr/bin/c++  -DusdTri_EXPORTS -I/home/michaelb/dev/USDPluginExamples/build/include -isystem /usr/include/python2.7  -O3 -DNDEBUG -fPIC   -o CMakeFiles/usdTri.dir/tokens.cpp.o -c /home/michaelb/dev/USDPluginExamples/src/usdTri/tokens.cpp
In file included from /home/michaelb/dev/USDPluginExamples/src/usdTri/tokens.cpp:6:
/home/michaelb/dev/USDPluginExamples/src/usdTri/./tokens.h:18:10: fatal error: pxr/pxr.h: No such file or directory
   18 | #include "pxr/pxr.h"
      |          ^~~~~~~~~~~
compilation terminated.
make[2]: *** [src/usdTri/CMakeFiles/usdTri.dir/build.make:83: src/usdTri/CMakeFiles/usdTri.dir/tokens.cpp.o] Error 1
make[2]: Leaving directory '/home/michaelb/dev/USDPluginExamples/build'
make[1]: *** [CMakeFiles/Makefile2:224: src/usdTri/CMakeFiles/usdTri.dir/all] Error 2
make[1]: Leaving directory '/home/michaelb/dev/USDPluginExamples/build'
make: *** [Makefile:150: all] Error 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
-# Minimum CMake version aligned with release branch of USD.
-cmake_minimum_required(VERSION 3.12)
+# 3.13 for target_link_directories.
+cmake_minimum_required(VERSION 3.13)
 
 # Define project.
 project(

--- a/README.md
+++ b/README.md
@@ -32,7 +32,7 @@ There are many other USD plugins available online - check out [USD Working Group
 
 The following dependencies are required:
 - C++ compiler
-- [CMake](https://cmake.org/documentation/) (3.12 or greater)
+- [CMake](https://cmake.org/documentation/) (3.13 or greater)
 - [USD](https://github.com/pixaranimationstudios/USD)
 - [Boost](https://boost.org) and [Intel TBB](https://www.threadingbuildingblocks.org/) (USD dependencies)
 

--- a/cmake/USDPluginTools.cmake
+++ b/cmake/USDPluginTools.cmake
@@ -179,6 +179,13 @@ function(_usd_library NAME)
             ${PYTHON_INCLUDE_DIR}
             ${Boost_INCLUDE_DIR}
             ${TBB_INCLUDE_DIRS}
+            ${USD_INCLUDE_DIR}
+    )
+
+    # Set-up library search path.
+    target_link_directories(${NAME}
+        PRIVATE
+            ${USD_LIBRARY_DIR}
     )
 
     # Link to libraries.
@@ -296,6 +303,12 @@ function(_usd_library NAME)
                     MFB_PACKAGE_NAME=${NAME}
                     MFB_ALT_PACKAGE_NAME=${NAME}
                     MFB_PACKAGE_MODULE=${PYTHON_MODULE_NAME}
+            )
+
+            # Set-up library search path.
+            target_link_directories(${PYLIB_NAME}
+                PRIVATE
+                    ${USD_LIBRARY_DIR}
             )
 
             # Apply link dependencies.

